tests/unit/dataactvalidator/test_d21_detached_award_financial_assistance.py

```diff
@@ -0,0 +1,35 @@
+from tests.unit.dataactcore.factories.staging import DetachedAwardFinancialAssistanceFactory
+from dataactcore.models.domainModels import SubTierAgency, CGAC
+from tests.unit.dataactvalidator.utils import number_of_errors, query_columns
+
+_FILE = 'd21_detached_award_financial_assistance'
+
+
+def test_column_headers(database):
+    expected_subset = {'row_number', 'funding_sub_tier_agency_co'}
+    actual = set(query_columns(_FILE, database))
+    assert expected_subset == actual
+
+
+def test_success(database):
+    """ FundingSubTierAgencyCode is an optional field, but when provided
+    must be a valid 4-digit sub-tier agency code.  """
+
+    subcode = SubTierAgency(sub_tier_agency_code='0000', cgac_id='1')
+    cgac = CGAC(cgac_id='1', cgac_code='001', agency_name='test')
```

Comment:
is cgac needed here or does the test work fine without it? I actually don't know, I'm legitimately asking.